gtsfm/two_view_estimator.py

```diff
@@ -218,16 +240,83 @@ def check_for_degeneracy(
         return i2Ri1, i2Ui1, v_corr_idxs, two_view_report
 
 
+def ground_truth_verifier(
+    keypoints_i1: Keypoints,
+    keypoints_i2: Keypoints,
+    matches_i1i2: np.ndarray,
+    intrinsics_i1: Cal3Bundler,
+    intrinsics_i2: Cal3Bundler,
+    dist_threshold: float,
+    gt_wTi1: Pose3,
+    gt_wTi2: Pose3,
+    gt_scene_mesh: trimesh.Trimesh,
+) -> Tuple[Optional[Rot3], Optional[Unit3], np.ndarray, float]:
+    """Verify correspondences and compute relative pose using ground truth data."""
+    logger.info("ground truth verifier")
+    # Compute ground truth correspondences.
+    gt_camera_i1 = PinholeCameraCal3Bundler(gt_wTi1, intrinsics_i1)
+    gt_camera_i2 = PinholeCameraCal3Bundler(gt_wTi2, intrinsics_i2)
+    inlier_mask, _ = metric_utils.mesh_inlier_correspondences(
+        keypoints_i1.extract_indices(matches_i1i2[:, 0]),
+        keypoints_i2.extract_indices(matches_i1i2[:, 1]),
+        gt_camera_i1,
+        gt_camera_i2,
+        gt_scene_mesh,
+        dist_threshold,
+    )
+    inlier_idxs = np.where(inlier_mask.ravel() == 1)[0]
+    v_corr_idxs = matches_i1i2[inlier_idxs]
+    logger.info("computed ground truth correspondences")
+
+    if v_corr_idxs.shape[0] < 15:
+        return None, None, np.array([], dtype=np.uint64), 0.0
+
+    # Compute essential matrix.
+    i2Ei1, _ = cv2.findEssentialMat(
```

Comment:
I think we can cut this down by about 20 lines if we just call `Ransac.verify()` here:
https://github.com/borglab/gtsfm/blob/master/gtsfm/frontend/verifier/ransac.py#L62